File: crates/sdk/examples/connect.rs
//! Attach to a running `whycodes serve` and list sessions.
//!
//! ```bash
//! whycodes serve
//! cargo run -p whycodes-sdk --example connect
//! ```

use whycodes_sdk::WhyCodesClient;

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let addr = std::env::args()
        .nth(1)
        .unwrap_or_else(|| "127.0.0.1:3030".into());
    let client = WhyCodesClient::connect(&addr).await?;
    let health = client.health().await?;
    println!(
        "protocol={} version={} project={}",
        health.protocol, health.version, health.project
    );
    for s in client.list_sessions().await? {
        println!("  {}  {}", s.id, s.title);
    }
    Ok(())
}

File: sdk/typescript/examples/connect.mjs
#!/usr/bin/env node
/** Attach to a running `whycodes serve` and list sessions. */
import { WhyCodesClient } from "../dist/index.js";

const addr = process.argv[2] ?? "127.0.0.1:3030";
const client = await WhyCodesClient.connect(addr);
const health = await client.health();
console.log(`protocol=${health.protocol} version=${health.version} project=${health.project}`);
for (const s of await client.listSessions()) {
  console.log(`  ${s.id}  ${s.title}`);
}
